src/wallet/wallet2.cpp

```diff
@@ -13643,60 +13643,89 @@ std::string wallet2::decrypt_with_view_secret_key(const std::string &ciphertext,
   return decrypt(ciphertext, get_account().get_keys().m_view_secret_key, authenticated);
 }
 //----------------------------------------------------------------------------------------------------
-std::string wallet2::make_uri(const std::string &address, const std::string &payment_id, uint64_t amount, const std::string &tx_description, const std::string &recipient_name, std::string &error) const
+std::string wallet2::make_uri(std::vector<uri_data> data, const std::string &payment_id, const std::string &tx_description, std::string &error) const
 {
-  cryptonote::address_parse_info info;
-  if(!get_account_address_from_str(info, nettype(), address))
+  std::string addresses = "";
+  std::string amounts = "";
+  bool amounts_used = false;
+  std::string recipients = "";
+  bool recipients_used = false;
+  for (uri_data entry : data)
   {
-    error = std::string("wrong address: ") + address;
-    return std::string();
-  }
-
-  // we want only one payment id
-  if (info.has_payment_id && !payment_id.empty())
-  {
-    error = "A single payment id is allowed";
-    return std::string();
-  }
+    cryptonote::address_parse_info info;
+    if(!get_account_address_from_str(info, nettype(), entry.address))
+    {
+      error = std::string("wrong address: ") + entry.address;
+      return std::string();
+    }
 
-  if (!payment_id.empty())
-  {
-    crypto::hash pid32;
-    if (!wallet2::parse_long_payment_id(payment_id, pid32))
+    // we want only one payment id
+    if (info.has_payment_id && !payment_id.empty())
     {
-      error = "Invalid payment id";
+      error = "A single payment id is allowed";
       return std::string();
     }
-  }
 
-  std::string uri = "monero:" + address;
-  unsigned int n_fields = 0;
+    if (!addresses.empty())
+    {
+      addresses += ";";
+    }
+    addresses += entry.address;
+    
+    if (!amounts.empty())
+    {
+      amounts += ";";
+    }
+    if (entry.amount > 0)
+    {
+      amounts_used = true;
+    }
+    amounts += cryptonote::print_money(entry.amount);
 
-  if (!payment_id.empty())
-  {
-    uri += (n_fields++ ? "&" : "?") + std::string("tx_payment_id=") + payment_id;
+    if (!recipients.empty())
+    {
+      recipients += ";";
+    }
+    if (!entry.recipient_name.empty())
+    {
+      recipients_used = true;
+      recipients += epee::net_utils::conver_to_url_format(entry.recipient_name);
+    }
+    else
+    {
+      recipients += "NA";
```

Comment:
Should be no need to use NA filler to handle lists with empty values, which can be done like this: `recipient_name=me;him;;you`